### PR TITLE
BTS-997: split way to big testsuite into its natural chunks

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

This testsuite wouldn't scale well with timeouts etc. in one piece. Chop it into its bits into own file. 

- [x] :hankey: Bugfix

### CODE REVIEW

File: tests/js/server/aql/aql-graph-traverser-optimizeIn.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+var _ = require('lodash');
+const internal = require('internal');
+const db = internal.db;
+const errors = require('@arangodb').errors;
+const gm = require('@arangodb/general-graph');
+const vn = 'UnitTestVertexCollection';
+const en = 'UnitTestEdgeCollection';
+const isCluster = require('@arangodb/cluster').isCluster();
+const removeCost = require('@arangodb/aql-helper').removeCost;
```
Comment (hkernbach) [inline tests/js/server/aql/aql-graph-traverser-optimizeIn.js:44]: Only this testfile needs the removeCost method. All other files are not using it, but they do require it. 
Please check if that might be true for other requires here as well.

File: tests/js/server/aql/aql-graph-traverser-brokenGraph.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+const {assertEqual, assertTrue, assertFalse, fail} = jsunity.jsUnity.assertions;
+
+const internal = require('internal');
+const db = internal.db;
+const errors = require('@arangodb').errors;
+const gm = require('@arangodb/general-graph');
+const vn = 'UnitTestVertexCollection';
+const en = 'UnitTestEdgeCollection';
+
+
```
Comment (hkernbach) [inline tests/js/server/aql/aql-graph-traverser-brokenGraph.js:43]:
```suggestion


```

File: tests/js/server/aql/aql-graph-traverser-complexInterna.js [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+'use strict';
+
+const jsunity = require('jsunity');
+const {assertEqual, assertTrue, assertFalse, fail} = jsunity.jsUnity.assertions;
+
+const internal = require('internal');
+const db = internal.db;
+const errors = require('@arangodb').errors;
+
+
```
Comment (hkernbach) [inline tests/js/server/aql/aql-graph-traverser-complexInterna.js:40]:
```suggestion


```

File: tests/js/server/aql/aql-graph-traverser-exampleGraphs.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+const vn = 'UnitTestVertexCollection';
+const en = 'UnitTestEdgeCollection';
+
+
+
+
+
+
+
+
```
Comment (hkernbach) [inline tests/js/server/aql/aql-graph-traverser-exampleGraphs.js:49]:
```suggestion








```

Comment (hkernbach) [inline tests/js/server/aql/aql-graph-traverser-exampleGraphs.js:49]:
```suggestion
```

**Comment by hkernbach** [comment]
Build still failing. Apart from that code looks good (except those failures). 
What we should do in the near future. Include JS formatting rules and auto-format. Like we're doing it already for C++ via clang-format. In general we should to this for any language (e.g. Python etc.)

File: tests/js/server/aql/aql-graph-traverser-complexInterna.js (hunk 3 above)
Comment (dothebart) [reply tests/js/server/aql/aql-graph-traverser-complexInterna.js:40]: lol. cannot delete lines by sugetion ;)

**Review by hkernbach — APPROVED** [review]

**Pull request merged**